src/DGtal/geometry/helpers/ContourHelper.h

```diff
@@ -62,29 +63,63 @@ class ContourHelper
 public:
   
   /**
-   * Compute the barycenter of the 2D contour.  
+   * Compute the barycenter of a set of points.  
    * 
-   * @param aContour the vector containing the 2D contour coordinates.
+   * @param[in] aSet the vector containing the set of points.
    * @return the resulting mean point. 
    **/
-  template <typename TPoint>
+  template <typename TPoint> 
   static 
-  TPoint getMeanPoint(const std::vector<TPoint> & aContour);
+  DGtal::PointVector<TPoint::dimension, double> 
+  getBarycenter(const std::vector<TPoint> & aSet);
   
 
   /**
-   * Checks if a contour given as a sequence of point is clockwise oriented or not.
+   * Checks if a polygonal curve given as a sequence of point is clockwise oriented or not.
    *
-   * @param aContour the vector containing the 2D contour coordinates.
-   * @return 'true' if the contour is counter clockwise oriented, 'false' otherwise.
+   * @param[in] aCurve the vector containing the 2D polygonal curve.
+   * @return 'true' if the polygonal curve is counter clockwise oriented, 'false' otherwise.
    **/
-
   template <typename TPoint> 
   static
   bool
-  isCounterClockWise(const std::vector<TPoint> & aContour);
+  isCounterClockWise(const std::vector<TPoint> & aCurve);
+
+
+
+  /**
+   * @return the Freeman code associated to given points (0,1,2, or
+   * 3). If the direction between the two points differs from these
+   * directions it returns 8.
+   *
+   * @param[in] pt1  a first point.
+   * @param[in] pt2  a second point.
+   * 
+   **/
+  template<typename TPoint>
+  static 
+  unsigned int 
+  getFreemanCode4C(const TPoint &pt1, const TPoint &pt2);
+  
 
-   
+  /**
+   * Transforms an input contour into an 8 connected contour.  The
+   * contour is given from two iterators on points.  If the input
+   * sequence contains some points not 4-connected, they are
+   * are ignored (and the transformation continue on the remaining
+   * parts).
+   *
```

Comment:
to understand: if the first point is disconnected and the rest 4-connected, then only the first point is in the output, am I right ?

So maybe the method could be named differently, it returns the longest 8-connected curve in a greedy way.